### PR TITLE
mercury ui: animated homescreen label, change homescreen flow 

### PR DESCRIPTION
**Pull request by TychoVrahe**
This PR introduces animated homecreen/lockscreen label - label disappears after timeout and reappears on touch.

The change homescreen flows are tweaked a bit to better match designs - but notable Tap to Confirm screen is still missing when changing homescreen to custom image.

### CODE REVIEW

File: core/embed/rust/src/ui/component/cached_jpeg.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        geometry::{Offset, Point, Rect},
+        shape,
+        shape::{render_on_canvas, ImageBuffer, Renderer, Rgb565Canvas},
+    },
+};
+
+pub struct CachedJpeg {
+    area: Rect,
+    image_size: Offset,
+    image: BinaryData<'static>,
```
Comment (matejcik) [inline core/embed/rust/src/ui/component/cached_jpeg.rs:15]: looks unused

**Review by matejcik — COMMENTED** [review]
LGTM, I see no problems in code apart from that one unused image

either @obrusvit or @mmilata should have a look at `homescreen.rs` to see whether it's in line with "how we do things" in mercury UI

File: core/embed/rust/src/ui/component/cached_jpeg.rs (hunk 1 above)
Comment (TychoVrahe) [reply core/embed/rust/src/ui/component/cached_jpeg.rs]: removed https://github.com/trezor/trezor-firmware/commit/c067919830ea7217fe9bee1e49d330e7920a1dc6

File: core/embed/rust/src/ui/model_mercury/component/homescreen.rs [hunk 2]
```diff
@@ -114,41 +275,54 @@ impl Homescreen {
...
+    fn level_to_style(level: u8) -> (Color, Color) {
         match level {
-            3 => (theme::ORANGE_DARK, theme::ICON_COINJOIN),
-            2 => (theme::ORANGE_DARK, theme::ICON_MAGIC),
-            1 => (theme::ORANGE_DARK, theme::ICON_WARN),
-            _ => (theme::ORANGE_DARK, theme::ICON_WARN),
+            3 => (theme::GREEN_DARK, theme::GREEN_LIME),
+            2 => (theme::ORANGE_DARK, theme::ORANGE_LIGHT),
+            1 => (theme::ORANGE_DARK, theme::ORANGE_LIGHT),
+            _ => (theme::ORANGE_DARK, theme::ORANGE_LIGHT),
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/homescreen.rs:308]: nit
```suggestion
            _ => (theme::ORANGE_DARK, theme::ORANGE_LIGHT),
```

**Review by mmilata — APPROVED** [review]

File: core/embed/rust/src/ui/model_mercury/component/homescreen.rs (hunk 2 above)
Comment (TychoVrahe) [reply core/embed/rust/src/ui/model_mercury/component/homescreen.rs]: https://github.com/trezor/trezor-firmware/commit/1ff0ac2ebf67939e0e52b05f512864ac1cf551d2

Comment (TychoVrahe) [reply core/embed/rust/src/ui/model_mercury/component/homescreen.rs]: or here, after rebase https://github.com/trezor/trezor-firmware/commit/1ba2cf1c6f8d29ebacbfb343cf4334b2bf30b384

**Comment by bosomt** [comment]
**QA NOK**

https://github.com/trezor/trezor-firmware/issues/3981

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 24.6.2 (b319fb1a1e4e2fb85e398589dec064c26450bab0)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.6.2 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.0 regular (revision dd4671a5104952ef505d28d1f9e94d1484b4607a)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**